### PR TITLE
Closes #222 | Create dataset loader for CreoleRC

### LINKED ISSUE #222: Create dataset loader for CreoleRC
Dataloader name: `creole_rc/creole_rc.py`
DataCatalogue: http://seacrowd.github.io/seacrowd-catalogue/card.html?creole_rc

|     Dataset| creole_rc |
|-------------|---|
| Description | CreoleRC is a subset created by the CreoleVal paper. Relation classification (RC) aims to identify semantic associations between entities within a text, essential for applications like knowledge base completion and question answering. The dataset is sourced from Wikipedia and manually annotated. CreoleRC contains 5 creoles, but SEACrowd is interested specifically in the Chavacano subset. |
| Subsets | - |
| Languages | cbk |
| Tasks | Relation Extraction |
| License | Creative Commons Attribution Share Alike 4.0 (cc-by-sa-4.0) |
| Homepage | https://github.com/hclent/CreoleVal/tree/main/nlu/relation_classification |
| HF URL | - |
| Paper URL | https://arxiv.org/abs/2310.19567 |

### PR DESCRIPTION
**Pull request by zwenyu**
closes #222 

### Checkbox
- [x] Confirm that this PR is linked to the dataset issue.
- [x] Create the dataloader script `seacrowd/sea_datasets/my_dataset/my_dataset.py` (please use only lowercase and underscore for dataset naming).
- [x] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_SEACROWD_VERSION` variables.
- [x] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `SEACrowdConfig` for the source schema and one for a seacrowd schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [x] Confirm that your dataloader script passes the test suite run with `python -m tests.test_seacrowd seacrowd/sea_datasets/<my_dataset>/<my_dataset>.py`.
- [x] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

### CODE REVIEW

**Review by faridlazuarda — APPROVED** [review]
I have manually tested all the functionality and subset (including the relations structuring) and this looks good to me! Thank you for the contribution, @zwenyu.

**Comment by sabilmakbar** [comment]
Hi @zwenyu, to make the reviewing tidier, would you like to separate config changes into a different PR? (so you can be attributed for bonus points in introducing new tasks, too)

**Comment by zwenyu** [comment]
@sabilmakbar I've removed the config changes in constant.py to a separate PR #502.

File: seacrowd/sea_datasets/creole_rc/creole_rc.py [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+                    "document_id": str(i),
+                    "passages": [{"id": f"passage-{i}", "type": "text", "text": [csv_data[i][0]], "offsets": [[0, len(csv_data[i][0])]]}],
+                    "entities": [
+                        {"id": f"{i}-entity-{csv_data[i][3]}", "type": "text", "text": [csv_data[i][1]], "normalized": [], "offsets": [[offset_entity1, offset_entity1 + len(csv_data[i][1])]]},
+                        {"id": f"{i}-entity-{csv_data[i][4]}", "type": "text", "text": [csv_data[i][2]], "normalized": [], "offsets": [[offset_entity2, offset_entity2 + len(csv_data[i][2])]]},
+                    ],
+                    "events": [],
+                    "coreferences": [],
+                    "relations": [{"id": f"{i}-relation-{csv_data[i][5]}", "type": properties_desc[csv_data[i][5]], "arg1_id": f"{i}-entity-{csv_data[i][3]}", "arg2_id": f"{i}-entity-{csv_data[i][4]}"}],
+                }
```
Comment (sabilmakbar) [inline seacrowd/sea_datasets/creole_rc/creole_rc.py:195]: missing `normalized` on this schema that leads to error on my end. Not sure why @faridlazuarda's test works on his end, prob due to earlier version of `datasets` module, tho. mine tested on `datasets==2.16.1`

File: seacrowd/sea_datasets/creole_rc/creole_rc.py [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+                    "property": csv_data[i][5],
+                    "property_desc": properties_desc[csv_data[i][5]],
+                    "tokens": json_data[i]["tokens"],
+                    "edgeset_left": json_data[i]["edgeSet"]["left"],
+                    "edgeset_right": json_data[i]["edgeSet"]["right"],
+                    "edgeset_triple": json_data[i]["edgeSet"]["triple"],
+                }
+            elif self.config.schema == f"seacrowd_{self.SEACROWD_SCHEMA_NAME}":
+                offset_entity1 = csv_data[i][0].find(csv_data[i][1])
+                offset_entity2 = csv_data[i][0].find(csv_data[i][2])
```
Comment (sabilmakbar) [inline seacrowd/sea_datasets/creole_rc/creole_rc.py:182]: Can we put a sanity check if either of these values isn't found, leading to a -1 value? we can skip those rows if that's the case

**Review by sabilmakbar — CHANGES_REQUESTED** [review]

**Comment by zwenyu** [comment]
@sabilmakbar Thanks for the comments. I've pushed changes addressing the issues. Can you check if they are ok?

**Comment by sabilmakbar** [comment]
> @sabilmakbar Thanks for the comments. I've pushed changes addressing the issues. I didn't see error for normalized, using datasets version 2.17.1, but I've added normalized now. Can you check if they are ok?

Initially, I pointed to the missing field of `normalized` in the SEACrowd Schema of KB under the `relations` column, which is okay if being filled with an empty list. But since your new changes also fill them with appropriate values, it works (and even better)!

**Review by sabilmakbar — APPROVED** [review]
LGTM! Thanks a lot, @zwenyu and @faridlazuarda!

**Pull request merged**